src/main/AuthenticationResponse.ts

```diff
@@ -256,56 +231,66 @@ function extractPresentations(resOpts: AuthenticationResponseOpts) {
   };
 }
 
-async function createSIOPResponsePayload(
-  verifiedJwt: VerifiedAuthenticationRequestWithJWT,
-  resOpts: AuthenticationResponseOpts
-): Promise<AuthenticationResponsePayload> {
+async function createSIOPIDToken(verifiedJwt: VerifiedAuthenticationRequestWithJWT, resOpts: AuthenticationResponseOpts): Promise<IdToken> {
   assertValidResponseOpts(resOpts);
   if (!verifiedJwt || !verifiedJwt.jwt) {
     throw new Error(SIOPErrors.VERIFY_BAD_PARAMS);
   }
   const supportedDidMethods = verifiedJwt.payload.registration?.subject_syntax_types_supported?.filter((sst) =>
     sst.includes(SubjectSyntaxTypesSupportedValues.DID.valueOf())
   );
-
   const state = resOpts.state || getState(verifiedJwt.payload.state);
   const nonce = verifiedJwt.payload.nonce || resOpts.nonce || getNonce(state);
-  const registration = createDiscoveryMetadataPayload(resOpts.registration);
-
-  //https://sphereon.atlassian.net/browse/VDX-140
-  // *********************************************************************************
-  // todo We are missing a wrapper object. Actually the current object is the id_token
-  // *********************************************************************************
-
-  const { verifiable_presentations, vp_token } = extractPresentations(resOpts);
-  const authenticationResponsePayload: AuthenticationResponsePayload = {
+  const idToken: IdToken = {
     iss: ResponseIss.SELF_ISSUED_V2,
-    sub: resOpts.did,
     aud: verifiedJwt.payload.redirect_uri,
-    did: resOpts.did,
-    sub_type: supportedDidMethods.length && resOpts.did ? SubjectIdentifierType.DID : SubjectIdentifierType.JKT,
-    state,
-    nonce,
     iat: Date.now() / 1000,
     exp: Date.now() / 1000 + (resOpts.expiresIn || 600),
-    registration,
+    sub: resOpts.did,
+    auth_time: Date.now() / 1000,
+    nonce,
+    _vp_token: {
```

Comment:
Why are we all of a sudden always including a _vp_token. SIOPv2 is a seperate spec from OID4VP. We can also do SIOPv2 only. Next _vp_token can be used within the ID token, but you can also have a vp_token as sibling of an id token

Reply:
fixed